SDKBuildScripts/Update-PlayFabSdk.ps1

```diff
@@ -0,0 +1,293 @@
+    [AllowEmptyString()]
+    [string]$ApiSpecGitUrl,
+    [Parameter(ValueFromPipelineByPropertyName = $true)]
+    [string]$OutputPath = "..\..\sdks",
+    [Parameter(ValueFromPipelineByPropertyName = $true)]
+    [string]$TargetSource,
+    [Parameter(ValueFromPipelineByPropertyName = $true)]
+    [switch]$KeepSource,
+    [Parameter(ValueFromPipelineByPropertyName = $true)]
+    [switch]$NonNullable,
+    [Parameter(ValueFromPipelineByPropertyName = $true)]
+    [switch]$Beta
+)
+
+begin
+{
+    if(!(Get-Command node))
+    {
+        throw "You must have Node.js installed to generate the SDK"
+    }
+
+    $sdkTargetSrcMap = @{
+        "ActionScriptSDK" = "actionscript";
+        "Cocos2d-xSDK" = "cpp-cocos2dx";
+        "CSharpSDK" = "csharp";
+        "JavaSDK" = "java";
+        "JavaScriptSDK" = "javascript";
+        "LuaSDK" = "LuaSdk";
+        "NodeSDK" = "js-node";
+        "Objective_C_SDK" = "objc";
+        "PhpSDK" = "PhpSdk";
+        "PostmanCollection" = "postman";
+        "PythonSDK" = "PythonSdk";
+        "SdkTestingCloudScript" = "SdkTestingCloudScript";
+        "UnrealMarketplacePlugin" = "UnrealMarketplacePlugin";
+        "UnitySDK" = "unity-v2";
+        "WindowsSDK" = "windowssdk";
+        "XPlatCppSDK" = "xplatcppsdk";
+    }
+
+    $sdksPath = $OutputPath
+    if(![System.IO.Path]::IsPathRooted($sdksPath))
+    {
+        $sdksPath = Resolve-Path (Join-Path $PSScriptRoot $OutputPath)
+    }
+
+    if(!(Test-Path $sdksPath))
+    {
+        mkdir $sdksPath | Out-Null
+    }
+
+    if($PSCmdlet.ParameterSetName -eq "ApiSpecPfUrlCloudVertical")
+    {
+        if($Vertical -or $Cloud)
+        {
+            # If either cloud or vertical is not provided, we want to remove the leading or trailing dot.
+            $cloudVertical = "$Vertical.$Cloud".TrimEnd(".")
+            $ApiSpecPfUrl = "https://$($cloudVertical).playfabapi.com/apispec"
+        }
+    }
+
+    if($PSCmdlet.ParameterSetName -eq "ApiSpecPath")
+    {
+        $apiSpecSource = "-apiSpecPath"
+
+        if($ApiSpecPath -and $ApiSpecPath -ne "default")
+        {
+            if(!(Test-Path $ApiSpecPath))
+            {
+                throw "Unable to find ApiSpecPath '$ApiSpecPath'."
+            }
+
+            $apiSpecSource += " $ApiSpecPath"
+        }
+    }
+    elseif($PSCmdlet.ParameterSetName -eq "ApiSpecPfUrl")
+    {
+        $apiSpecSource = "-apiSpecPfUrl"
+
+        if($ApiSpecPfUrl -and $ApiSpecPfUrl -ne "default")
+        {
+            if(![Uri]::IsWellFormedUriString($ApiSpecPfUrl, "Absolute"))
+            {
+                throw "You must specify a valid URL for ApiSpecPfUrl.  Unable to parse '$ApiSpecPfUrl' as a URL."
+            }
+
+            $apiSpecSource += " $ApiSpecPfUrl"
+        }
+    }
+    else
+    {
+        $apiSpecSource = "-apiSpecGitUrl"
+
+        if($ApiSpecGitUrl -and $ApiSpecGitUrl -ne "default")
+        {
+            $apiSpecSource += " $ApiSpecGitUrl"
```

Comment:
You could check:
if(![Uri]::IsWellFormedUriString($ApiSpecGitUrl, "Absolute"))
here as well.